Formula/emacs-plus@28.rb

```diff
@@ -67,6 +68,14 @@ class EmacsPlusAT28 < EmacsBase
     depends_on "zlib" => :build
   end
 
+  if build.with? "native-comp-aot"
+    depends_on "libgccjit" => :recommended
+    depends_on "gcc" => :build
+    depends_on "gmp" => :build
+    depends_on "libjpeg" => :build
+    depends_on "zlib" => :build
+  end
+
```

Comment:
You can use `||` operator:

```suggestion
  if build.with? "native-comp" || build.with? "native-comp-aot"
    depends_on "libgccjit" => :recommended
    depends_on "gcc" => :build
    depends_on "gmp" => :build
    depends_on "libjpeg" => :build
    depends_on "zlib" => :build
  end
  
```

Reply:
I had considered it, but since it was a late-night idea, I wasn't sure 🥲 
I'll refactor these bits.

About having non-aot native compilation, perhaps it won't hurt users with very fast machines (e.g. Apple Silicon and Intel 10th+ gen processors), where background native compilation won't hurt operative performance.

For slower machines, AOT moving native compilation time to the build process gives native-comp of ELPA packages a bit of a break.

There may be as many use cases as Emacs users, to each their own reasons; to all, an extra option to decide according to their needs.

I'll get back to you soon with the proposed deduplications.

Reply:
I'd pretty much prefer one option, because it's easier to support.

In case we decide to keep two, they should be marked as mutually exclusive using odie. Like this:

https://github.com/d12frosted/homebrew-emacs-plus/blob/d5dd64d3ad95b5e99c282f8cb7c596ed69da108b/Formula/emacs-plus%4030.rb#L70-L74